magento/magento2
# #21737 Duplicating product with translated url keys over multiple sto…

…reviews causes non-unique url keys to be generated





### Description (*)

When a product is duplicated, only one URL key is used(from scope where was clicked the button "Save & Duplicate"), it is modified and set for all store views. I've added generation and saving unique URL keys from URL keys of the source product for all store views.
### Fixed Issues (if relevant)

1. magento/magento2#21737: Duplicating product with translated url keys over multiple storeviews causes non-unique url keys to be generated

### Manual testing scenarios (*)

1. Make sure you have 2 storeviews, I've opted for the names 'English' and 'French'
2. Create a category "Test"
3. Create a product:
    Name: Default
    Sku: Default
    Price: 123
    Qty: 9999
    Category: Test
    Url key: default
4. Switch to the English storeview, and change the url key of the product to 'english', uncheck 'Create Permanent Redirect for old URL', save changes
5. Switch to the French storeview, and change the url key of the product to 'french', uncheck 'Create Permanent Redirect for old URL', save changes
6. Back on the 'All Store Views' scope, open the product and click the dropdown next to the Save button and choose 'Save & Duplicate'
7. Take a look in the database at the various url keys and url rewrites which got generated, they all unique

### Contribution checklist (*)
